### PR TITLE
Move RpcServer's dependencies out

### PR DESCRIPTION
**Pull request by przepompownia**
The description below was mostly copied from https://github.com/phpvim/phpcd.vim/pull/32 where these first two commits were appended originally. 
I got the motivation when I trying to test the PR mentioned above. Daemon did not work with the newly installed PHP7 from Debian repo. The cause was the lack of msgpack extension but i discovered it after running PHPID manually from bash. 

I would like to create simple logger class that gives the same effect, especially uses code from your methods, but exists as a separate responsibility. It would be instantiated in the "controller", use there and passed to RpcServer as another dependency and used there. It would be good to make it as an implementation of `\Psr\Log\LoggerInterface` (http://www.php-fig.org/psr/psr-3/).

These first two commits contain moving `MessagePackUnpacker` creation outside but they mainly prepare the ground for avoiding code duplication in both "controllers" ("adapter" may be more accurate name).

### CODE REVIEW

**Comment by lvht** [comment]
Would you plan to add an Logger implementation in this PR?

**Comment by przepompownia** [comment]
Yes, I still prepare the space for it. Do you still insist on avoiding the use of phpcd's own autoloader? It would to avoid manual `require` each used class.

**Comment by przepompownia** [comment]
I hope that the recent commit will convince you to abandon the manual `require`ment. In this version the user does not have to remember to add the `-o` option to `install`, `update` or `dumpautoload` (implicitly enabled `optimize-autoloader`), but also the autoloader does not perform searching any class file that is not present in `./vendor/composer/autoload_classmap.php`.

**Comment by przepompownia** [comment]
@lvht, @imaben what do you think about merging this version into master? 

I would like to add (maybe configurable) datetime stamps to log lines but not necessary in this PR.

File: composer.json [hunk 1]
```diff
@@ -7,9 +7,12 @@
...
     ],
     "autoload": {
-        "classmap": [
-            "php"
-        ]
+        "psr-4": {"PHPCD\\": "php/"}
     },
-    "require": {}
+    "require": {
+        "psr/log": "dev-master"
```
Comment (lvht) [inline composer.json]: Why not dependent the `psr/log 1.0.0`?

**Comment by lvht** [comment]
@przepompownia Please update the install instruction. Because we need the composer to install and autoload the dependences..

**Comment by przepompownia** [comment]
Both Readme files already contain hint about using `composer update` first.

**Comment by lvht** [comment]
the `composer update` in the readme file is not update the phpcd but the actual project you work on.

**Comment by przepompownia** [comment]
By the way, I changed suggestion for project directory. Some projects versioning composer.lock to keep control of the dependencies version.

Could you translate the Chinese version? I would not like to paste raw content from any web translator.

**Comment by przepompownia** [comment]
@lvht Do you have something against merging this PR into master?

**Pull request merged**